File: Js_Solutions/max_char/max_char.js
// Given a string, return a character
// that is most used in the string

function maxChar(str) {
    let charMap = {};
    let max = 0;
    let maxCh = '';

    for (let char of str) {
        if (charMap[char]) {
            charMap[char]++;
        } else {
            charMap[char] = 1;
        }
    }

    for (let char in charMap) {
        if (charMap[char] > max) {
            max = charMap[char];
            maxCh = char;
        }
    }

    return maxCh;
}
console.log(maxChar('Hello everyone'));

File: Python_Solutions/max_char/max_char.py
# Given a string, return a character
# that is most used in the string

def maxChar(string):
    charMap = dict()
    max_Occurrance = 0
    max_Ch = ''

    for ch in string:
        if ch not in charMap:
            charMap[ch] = 1
        else:
            charMap[ch] += 1

    for ch in charMap:
        if charMap[ch] > max_Occurrance:
            max_Occurrance = charMap[ch]
            max_Ch = ch

    return max_Ch


print(maxChar('Hello everyone'))
